transformer_engine/jax/cpp_extensions.py

```diff
@@ -2228,45 +2259,53 @@ class CrossFusedAttnBwdPrimitive(BasePrimitive):
     """
     name = "te_cross_fused_attn_backward"
     multiple_results = True
-    impl_static_args = (6, 7, 8, 9, 10)
+    impl_static_args = (8, 9, 10, 11, 12)
     inner_primitive = None
     outer_primitive = None
 
     @staticmethod
-    def abstract(q_aval, kv_aval, softmax_aux_aval, doutput_aval, q_cu_seqlen_aval,
-                 kv_cu_seqlen_aval, *, attn_bias_type, attn_mask_type, scaling_factor,
-                 dropout_probability, is_training):
+    def abstract(q_aval, kv_aval, softmax_aux_aval, rng_state_aval, output_aval, doutput_aval,
+                 q_cu_seqlen_aval, kv_cu_seqlen_aval, *, attn_bias_type, attn_mask_type,
+                 scaling_factor, dropout_probability, is_training):
         """
         Cross fused attention bwd abstract
         """
-        del attn_bias_type, attn_mask_type
-        del scaling_factor, dropout_probability, is_training
+        del softmax_aux_aval, rng_state_aval, output_aval
+        del attn_mask_type, scaling_factor, dropout_probability, is_training
         q_dtype = dtypes.canonicalize_dtype(q_aval.dtype)
         kv_dtype = dtypes.canonicalize_dtype(kv_aval.dtype)
-        softmax_aux_dtype = dtypes.canonicalize_dtype(softmax_aux_aval.dtype)
+
         doutput_dtype = dtypes.canonicalize_dtype(doutput_aval.dtype)
-        assert q_dtype == kv_dtype == softmax_aux_dtype == doutput_dtype
-        # outer_primitve is squeezed_mask, inner_primitive is cu_seqlen
+        assert q_dtype == kv_dtype == doutput_dtype
         assert q_cu_seqlen_aval.dtype == kv_cu_seqlen_aval.dtype
 
+        *batch_shape, q_max_seqlen, q_num_head, _ = q_aval.shape
+        kv_max_seqlen = kv_aval.shape[-4]
+        if attn_bias_type == NVTE_Bias_Type.NVTE_NO_BIAS:
+            bias_shape = (0,)
+        else:
+            bias_shape = (*batch_shape[:-1], 1, q_num_head, q_max_seqlen, kv_max_seqlen)
+        bias_dtype = q_dtype
+
         dq_aval = q_aval.update(shape=q_aval.shape, dtype=q_dtype)
         dkv_aval = kv_aval.update(shape=kv_aval.shape, dtype=kv_dtype)
-        return dq_aval, dkv_aval
+        dbias_aval = q_aval.update(shape=bias_shape, dtype=bias_dtype)
```

Comment:
Should this be bias_aval.update?

Reply:
There is no bias_aval during the backward (calculating the dbias doesn't require the bias). So this is a trick and intended, after `.update`, the `dbias_aval` will have the correct shape and type info.